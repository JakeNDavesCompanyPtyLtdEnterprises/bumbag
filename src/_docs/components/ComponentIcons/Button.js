import React from 'react';

const SvgComponent = props => (
  <svg width={72} height={30} fill="none" {...props}>
    <path
      d="M69.248 1H3.202C1.986 1 1 2.527 1 4.412v21.323c0 1.885.986 3.412 2.202 3.412h66.046c1.216 0 2.202-1.527 2.202-3.412V4.412C71.45 2.527 70.464 1 69.248 1z"
      fill="#444bc9"
      stroke="#08175D"
    />
    <path d="M64.02 13.368H8.98v4.264h55.04v-4.264z" fill="#fff" />
  </svg>
);

export default SvgComponent;
