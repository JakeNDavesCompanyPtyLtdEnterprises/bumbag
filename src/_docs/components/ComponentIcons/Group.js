import React from 'react';

const SvgComponent = props => (
  <svg width={139} height={30} fill="none" {...props}>
    <path d="M4 .5h65.5v29H4A3.5 3.5 0 0 1 .5 26V4A3.5 3.5 0 0 1 4 .5z" fill="#444bc9" stroke="#000" />
    <path d="M134.5 29.5H69V.5h65.5A3.5 3.5 0 0 1 138 4v22a3.5 3.5 0 0 1-3.5 3.5z" stroke="#000" />
    <path d="M132.539 13H77.5v4.265h55.039V13z" fill="#444bc9" />
    <path d="M61.539 13H6.5v4.265h55.039V13z" fill="#fff" />
  </svg>
);

export default SvgComponent;
