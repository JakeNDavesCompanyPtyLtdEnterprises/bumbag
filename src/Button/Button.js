// @flow
import React, { Fragment, type Node } from 'react';
import styled from 'reakit/styled';
import Box from 'reakit/Box';

import type { ButtonType, Palette, Size } from '../types';
import Spinner from '../Spinner';
import _Button, { ButtonIcon } from './styled';

const Text = styled.span`
  align-items: center;
  display: inline-flex;
`;
export const SpinnerWrapper = styled(Box)`
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;

  & + ${Text} {
    opacity: 0;
  }
`;

type Props = {
  use?: any,
  children: Node,
  className?: string,
  /** Makes the button disabled. The user is unable to interact with the button. */
  disabled?: boolean,
  /** Icon that appears on the right side of the button. */
  iconAfter?: boolean,
  /** Icon that appears on the left side of the button. */
  iconBefore?: boolean,
  /** Adds a loading indicator to the button. */
  isLoading?: boolean,
  /** Makes the button not interactable. */
  isStatic?: boolean,
  kind?: ButtonType,
  palette?: Palette,
  size?: Size
};

export const Button = ({
  children,
  className,
  disabled,
  iconAfter,
  iconBefore,
  isLoading,
  isStatic,
  kind,
  palette,
  size,
  ...props
}: Props) => {
  const child = (
    <Fragment>
      {iconBefore && <ButtonIcon icon={iconBefore} isBefore />}
      {children}
      {iconAfter && <ButtonIcon icon={iconAfter} isAfter />}
    </Fragment>
  );
  return (
    <_Button
      className={className}
      disabled={disabled}
      isLoading={isLoading}
      isStatic={isStatic}
      kind={kind}
      palette={palette}
      size={size}
      {...props}
    >
      {isLoading ? (
        <Fragment>
          <SpinnerWrapper>
            <Spinner color={kind === 'default' ? `${palette || ''}Inverted` : palette} />
          </SpinnerWrapper>
          <Text>{child}</Text>
        </Fragment>
      ) : (
        <Fragment>{child}</Fragment>
      )}
    </_Button>
  );
};

Button.defaultProps = {
  use: undefined,
  className: undefined,
  disabled: false,
  iconAfter: undefined,
  iconBefore: undefined,
  isLoading: false,
  isStatic: false,
  kind: 'default',
  palette: 'default',
  size: 'default'
};

export default Button;
