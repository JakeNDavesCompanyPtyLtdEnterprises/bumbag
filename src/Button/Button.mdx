---
name: Button
route: /components/button
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Button from './index';
import { Button as ButtonProps } from './Button';
import { ActionButtons } from './ActionButtons';
import Group from '../Group';
import Set from '../Set';

# Button

## Import

`import { Button } from 'fannypack'`

## Basic Usage

<Playground>
  <Button>Default button</Button>
</Playground>

## Custom elements

Button, by default, renders a `<button>` element.
However, you are able to change it to a custom element using the `as` prop. See more [here](https://reakit.io/components/button).

<Playground>
  <Set>
    <Button>Default</Button>
    <Button use="a" href="#">
      Anchor
    </Button>
  </Set>
</Playground>

## Colors

Colors can be added to a button with the `palette` prop.

<Playground>
  <Set>
    <Button>Default</Button>
    <Button palette="primary">Primary</Button>
    <Button palette="secondary">Secondary</Button>
    <Button palette="success">Success</Button>
    <Button palette="danger">Danger</Button>
    <Button palette="warning">Warning</Button>
  </Set>
</Playground>

## Loading

To indicate a button is loading, use the `isLoading` prop.

<Playground>
  <Set>
    <Button isLoading>Default</Button>
    <Button palette="primary" isLoading>
      Primary
    </Button>
    <Button palette="primary" isLoading kind="link">
      Primary
    </Button>
    <Button palette="success" isLoading kind="outlined">
      Success
    </Button>
  </Set>
</Playground>

## Outlines

An outline can be placed on a button by assigning `"outlined"` to the `kind` prop.

<Playground>
  <Set>
    <Button palette="primary" kind="outlined">
      Primary
    </Button>
    <Button palette="success" kind="outlined">
      Success
    </Button>
    <Button palette="danger" kind="outlined">
      Danger
    </Button>
  </Set>
</Playground>

## Sizes

Buttons are available in four different sizes and can be added with a `size` prop.

<Playground>
  <Set>
    <Button size="small">Small</Button>
    <Button>Default</Button>
    <Button size="medium">Medium</Button>
    <Button size="large">Large</Button>
  </Set>
</Playground>

## Links

A button can appear as a link with no styling by assigning `"link"` to the `type` prop.

<Playground>
  <Button palette="primary" kind="link">
    Default
  </Button>
</Playground>

## Static

To make a button non-interactable, add the `isStatic` prop.

<Playground>
  <Button isStatic>Static</Button>
</Playground>

## Disabled

To indicate a button as disabled, use the `disabled` prop.

<Playground>
  <Set>
    <Button disabled>Default</Button>
    <Button palette="primary" disabled>
      Primary
    </Button>
    <Button palette="success" disabled>
      Success
    </Button>
    <Button palette="danger" disabled>
      Danger
    </Button>
    <Button palette="warning" disabled>
      Warning
    </Button>
  </Set>
</Playground>

## Sets

To display a set of buttons, use the [`<Set>` component](/components/set).

<Playground>
  <Set>
    <Button>Previous</Button>
    <Button>Next</Button>
  </Set>
</Playground>

## Group

To group a set of buttons together, use the `<Group>` component.

<Playground>
  <Group>
    <Button>Previous</Button>
    <Button>Now</Button>
    <Button>Next</Button>
  </Group>
  <Group isVertical marginTop="xsmall">
    <Button>Previous</Button>
    <Button>Now</Button>
    <Button>Next</Button>
  </Group>
</Playground>

## `<Button>` Props

<PropsTable of={ButtonProps} />

## Theming

### Schema

```jsx
{
  base: string | Object,

  // disabled button type styles
  disabled: string | Object,

  // link button type styles
  link: string | Object,

  // loading button styles
  loading: string | Object,

  // static button styles
  static: stirng | Object,

  // size styles
  sizes: {
    small: string | Object,
    medium: string | Object,
    large: string | Object
  }
}
```

# ActionButtons

## Import

`import { ActionButtons } from 'fannypack'`

## Basic Usage

The `<ActionButtons>` component renders a primary action button (e.g 'Submit'), as well as a secondary action button (e.g 'Cancel').
They can be helpful for forms, dialogs and other things that require an action.

<Playground>
  <ActionButtons onClickSubmit={() => console.log('submitted')} onClickCancel={() => console.log('cancelled')} />
</Playground>

## Colors

Action buttons can also take a color from the palette using the `palette` prop.

<Playground>
  <Set isVertical>
    <ActionButtons
      palette="text"
      onClickSubmit={() => console.log('submitted')}
      onClickCancel={() => console.log('cancelled')}
    />
    <ActionButtons
      palette="secondary"
      onClickSubmit={() => console.log('submitted')}
      onClickCancel={() => console.log('cancelled')}
    />
    <ActionButtons
      palette="danger"
      onClickSubmit={() => console.log('submitted')}
      onClickCancel={() => console.log('cancelled')}
    />
  </Set>
</Playground>

## Custom text

To modify the text of the buttons, use the `submitText` and/or `cancelText` props.

<Playground>
  <ActionButtons cancelText="Close" submitText="Save" onClickSubmit={() => console.log('submitted')} onClickCancel={() => console.log('cancelled')} />
</Playground>

## Loading

To indicate the primary button as loading, use the `isLoading` prop.

<Playground>
  <ActionButtons isLoading onClickSubmit={() => console.log('submitted')} onClickCancel={() => console.log('cancelled')} />
</Playground>

## Type

To change the type of the submit button (e.g `submit` or `reset`). Use the `type` prop.

<Playground>
  <ActionButtons type="submit" />
</Playground>

## Custom props

You can pass custom button props to action buttons using `submitProps` or `cancelProps`.

<Playground>
  <ActionButtons cancelText="Reject" submitProps={{ disabled: true }} cancelProps={{ palette: 'danger' }} />
</Playground>

## `<ActionButtons>` Props

<PropsTable of={ActionButtons} />
