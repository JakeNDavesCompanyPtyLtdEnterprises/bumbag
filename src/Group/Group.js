// @flow
import React, { type Node } from 'react';
import { withTheme } from 'reakit/styled';
import _get from 'lodash/get';
import { type Breakpoint } from '../typesold';
import _Group from './styled';

type Props = {
  children: Node,
  className?: string,
  isVertical?: boolean,
  theme?: Object,
  verticalAt?: Breakpoint | number
};

const getVerticalBreakpoint = ({ breakpoint, theme }: { breakpoint: Breakpoint | number, theme?: Object }) => {
  return typeof breakpoint === 'number' ? breakpoint : _get(theme, `fannypack.layout[${breakpoint}Breakpoint]`);
};

export const Group = ({ children, className, isVertical, theme, verticalAt: _verticalAt, ...props }: Props) => {
  const verticalAt = getVerticalBreakpoint({ breakpoint: _verticalAt, theme });
  return (
    <_Group className={className} vertical={isVertical} verticalAt={verticalAt} {...props}>
      {children}
    </_Group>
  );
};

Group.defaultProps = {
  className: undefined,
  isVertical: false,
  theme: {},
  verticalAt: undefined
};

export default withTheme(Group);
