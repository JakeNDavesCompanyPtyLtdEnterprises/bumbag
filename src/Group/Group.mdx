---
name: Group
route: /utils/group
menu: Utilities
---

import { Playground, PropsTable } from 'docz';
import {Group} from './Group';
import { Box, Block, Flex } from '../primitives';
import Button from '../Button';

# Group

## Import

`import { Group } from 'fannypack'`

## Basic Usage

The `<Group>` component groups components together with their respective `border-radius`.

<Playground>
  <Group>
    <Box backgroundColor="primary" color="white" padding="0.5rem 1rem">
      Hello
    </Box>
    <Button>Send</Button>
  </Group>
</Playground>

## Vertical

A group can also be aligned vertically.

<Playground>
  <Group isVertical>
    <Box backgroundColor="primary" color="white" padding="0.5rem 1rem" textAlign="center">
      Hello
    </Box>
    <Button>Send</Button>
  </Group>
</Playground>

## Responsive

A group can snap from horizontal to vertical after a given breakpoint with the `verticalAt` prop.

The `verticalAt` prop accepts the following values: `"mobile"`, `"tablet"`, `"desktop"`, `"widescreen"` or `"fullHD"`.

<Playground>
  <Group verticalAt="tablet">
    <Box backgroundColor="primary" color="white" padding="0.5rem 1rem">
      Hello
    </Box>
    <Button>Send</Button>
  </Group>
</Playground>

## Props

<PropsTable of={Group} />

## Theming

### Schema

```jsx
{
  base: string | Object
}
```
