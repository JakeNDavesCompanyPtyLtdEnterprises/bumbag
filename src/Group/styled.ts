import { Group } from 'reakit';
import { theme } from 'styled-tools';

import styled, { css } from '../styled';
import { GroupProps } from './Group';

export default styled(Group)<GroupProps>`
  & > * {
    border-radius: 4px;
  }

  & input:focus,
  & select:focus {
    z-index: 1;
    position: relative;
  }

  & > *:first-child {
    & input,
    & select {
      ${props =>
        props.vertical
          ? css`
              border-bottom-right-radius: 0;
              border-bottom-left-radius: 0;
            `
          : css`
              border-bottom-right-radius: 0;
              border-top-right-radius: 0;
            `};
    }
  }

  & > *:last-child {
    & input,
    & select {
      ${props =>
        props.vertical
          ? css`
              border-top-left-radius: 0;
              border-top-right-radius: 0;
            `
          : css`
              border-top-left-radius: 0;
              border-bottom-left-radius: 0;
            `};
    }
  }

  & > *:not(:first-child):not(:last-child) > input,
  & > *:not(:first-child):not(:last-child) > select {
    border-radius: 0;
  }

  & {
    ${theme('fannypack.Group.base')};
  }
`;
