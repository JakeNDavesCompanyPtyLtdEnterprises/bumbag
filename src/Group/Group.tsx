import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { GroupProps as ReakitGroupProps } from 'reakit/ts';
// @ts-ignore
import _get from 'lodash/get';
import { withTheme } from '../styled';
import { Omit, Breakpoint, breakpointPropType } from '../types';
import _Group from './styled';

export interface LocalGroupProps {
  children: React.ReactNode;
  className?: string;
  isVertical?: boolean;
  theme?: Object;
  verticalAt?: Breakpoint | number;
}
export type GroupProps = LocalGroupProps & Omit<ReakitGroupProps, 'verticalAt'>;

const getVerticalBreakpoint = ({ breakpoint, theme }: { breakpoint?: Breakpoint | number; theme?: Object }) => {
  return typeof breakpoint === 'number' ? breakpoint : _get(theme, `fannypack.layout[${breakpoint}Breakpoint]`);
};

export const Group: React.FunctionComponent<LocalGroupProps> = ({
  children,
  className,
  isVertical,
  theme,
  verticalAt: _verticalAt,
  ...props
}) => {
  const verticalAt = getVerticalBreakpoint({ breakpoint: _verticalAt, theme });
  return (
    <_Group className={className} vertical={isVertical} verticalAt={verticalAt} {...props}>
      {children}
    </_Group>
  );
};

Group.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  isVertical: PropTypes.bool,
  theme: PropTypes.object,
  verticalAt: PropTypes.oneOfType([breakpointPropType, PropTypes.number])
};
Group.defaultProps = {
  className: undefined,
  isVertical: false,
  theme: {},
  verticalAt: undefined
};

// @ts-ignore
const C: React.FunctionComponent<GroupProps> = withTheme(Group);
export default C;
