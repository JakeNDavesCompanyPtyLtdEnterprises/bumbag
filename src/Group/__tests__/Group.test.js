import React from 'react';
import render from '../../_utils/tests/render';
import Button from '../../Button/Button.tsx';
import Group from '../Group';
import Box from 'reakit/Box';
it('renders correctly for a basic group', () => {
  const { container } = render(
    <Group>
      <Box backgroundColor="primary" color="white" padding="0.5rem 1rem">
        Hello
      </Box>
      <Button>Send</Button>
    </Group>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a vertical group', () => {
  const { container } = render(
    <Group isVertical>
      <Box backgroundColor="primary" color="white" padding="0.5rem 1rem">
        Hello
      </Box>
      <Button>Send</Button>
    </Group>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a responsive group', () => {
  const { container } = render(
    <Group verticalAt="tablet">
      <Box backgroundColor="primary" color="white" padding="0.5rem 1rem">
        Hello
      </Box>
      <Button>Send</Button>
    </Group>
  );
  expect(container.firstChild).toMatchSnapshot();
});
