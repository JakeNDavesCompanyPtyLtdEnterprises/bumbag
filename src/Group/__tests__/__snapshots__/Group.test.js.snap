// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a basic group 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c4 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  color: white !important;
  background-color: #3926a5 !important;
}

.c6 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #bfbfbf;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c5[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c5:hover {
  background-color: #efefef;
}

.c5:hover:active {
  background-color: #e3e3e3;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 > *:not(:first-child):not(:last-child),
.c3 > *:not(:first-child):not(:last-child) .GroupItem-sc-2orogz-0 {
  border-radius: 0;
  border-left-width: 0;
}

.c3 > *:first-child,
.c3 > *:first-child .GroupItem-sc-2orogz-0 {
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}

.c3 > *:last-child,
.c3 > *:last-child .GroupItem-sc-2orogz-0 {
  border-top-left-radius: 0;
  border-left-width: 0;
  border-bottom-left-radius: 0;
}

.c2 > * {
  border-radius: 4px;
}

.c2 input:focus,
.c2 select:focus {
  z-index: 1;
  position: relative;
}

.c2 > *:first-child input,
.c2 > *:first-child select {
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}

.c2 > *:last-child input,
.c2 > *:last-child select {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.c2 > *:not(:first-child):not(:last-child) > input,
.c2 > *:not(:first-child):not(:last-child) > select {
  border-radius: 0;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c1"
    role="group"
  >
    <div
      class="c4"
      color="white"
      style="color: white; padding: 0.5rem 1rem;"
    >
      Hello
    </div>
    <button
      class="c5 c6"
      kind="default"
      style="size: default;"
    >
      <span
        class="c7"
      >
        Send
      </span>
    </button>
  </div>
</div>
`;

exports[`renders correctly for a responsive group 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c4 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  color: white !important;
  background-color: #3926a5 !important;
}

.c6 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #bfbfbf;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c5[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c5:hover {
  background-color: #efefef;
}

.c5:hover:active {
  background-color: #e3e3e3;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 > *:not(:first-child):not(:last-child),
.c3 > *:not(:first-child):not(:last-child) .GroupItem-sc-2orogz-0 {
  border-radius: 0;
}

.c3 > *:first-child,
.c3 > *:first-child .GroupItem-sc-2orogz-0 {
  border-bottom-right-radius: 0;
}

.c3 > *:last-child,
.c3 > *:last-child .GroupItem-sc-2orogz-0 {
  border-top-left-radius: 0;
}

.c2 > * {
  border-radius: 4px;
}

.c2 input:focus,
.c2 select:focus {
  z-index: 1;
  position: relative;
}

.c2 > *:first-child input,
.c2 > *:first-child select {
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}

.c2 > *:last-child input,
.c2 > *:last-child select {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.c2 > *:not(:first-child):not(:last-child) > input,
.c2 > *:not(:first-child):not(:last-child) > select {
  border-radius: 0;
}

@media (max-width:768px) {
  .c3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:769px) {
  .c3 > *:not(:first-child):not(:last-child),
  .c3 > *:not(:first-child):not(:last-child) .GroupItem-sc-2orogz-0 {
    border-left-width: 0;
  }
}

@media (max-width:768px) {
  .c3 > *:not(:first-child):not(:last-child),
  .c3 > *:not(:first-child):not(:last-child) .GroupItem-sc-2orogz-0 {
    border-top-width: 0;
  }
}

@media (min-width:769px) {
  .c3 > *:first-child,
  .c3 > *:first-child .GroupItem-sc-2orogz-0 {
    border-top-right-radius: 0;
  }
}

@media (max-width:768px) {
  .c3 > *:first-child,
  .c3 > *:first-child .GroupItem-sc-2orogz-0 {
    border-bottom-left-radius: 0;
  }
}

@media (min-width:769px) {
  .c3 > *:last-child,
  .c3 > *:last-child .GroupItem-sc-2orogz-0 {
    border-left-width: 0;
    border-bottom-left-radius: 0;
  }
}

@media (max-width:768px) {
  .c3 > *:last-child,
  .c3 > *:last-child .GroupItem-sc-2orogz-0 {
    border-top-width: 0;
    border-top-right-radius: 0;
  }
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c1"
    role="group"
  >
    <div
      class="c4"
      color="white"
      style="color: white; padding: 0.5rem 1rem;"
    >
      Hello
    </div>
    <button
      class="c5 c6"
      kind="default"
      style="size: default;"
    >
      <span
        class="c7"
      >
        Send
      </span>
    </button>
  </div>
</div>
`;

exports[`renders correctly for a vertical group 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c4 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  color: white !important;
  background-color: #3926a5 !important;
}

.c6 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #bfbfbf;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c5[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c5:hover {
  background-color: #efefef;
}

.c5:hover:active {
  background-color: #e3e3e3;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 > *:not(:first-child):not(:last-child),
.c3 > *:not(:first-child):not(:last-child) .GroupItem-sc-2orogz-0 {
  border-radius: 0;
  border-top-width: 0;
}

.c3 > *:first-child,
.c3 > *:first-child .GroupItem-sc-2orogz-0 {
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
}

.c3 > *:last-child,
.c3 > *:last-child .GroupItem-sc-2orogz-0 {
  border-top-left-radius: 0;
  border-top-width: 0;
  border-top-right-radius: 0;
}

.c2 > * {
  border-radius: 4px;
}

.c2 input:focus,
.c2 select:focus {
  z-index: 1;
  position: relative;
}

.c2 > *:first-child input,
.c2 > *:first-child select {
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
}

.c2 > *:last-child input,
.c2 > *:last-child select {
  border-top-left-radius: 0;
  border-top-right-radius: 0;
}

.c2 > *:not(:first-child):not(:last-child) > input,
.c2 > *:not(:first-child):not(:last-child) > select {
  border-radius: 0;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c1"
    role="group"
  >
    <div
      class="c4"
      color="white"
      style="color: white; padding: 0.5rem 1rem;"
    >
      Hello
    </div>
    <button
      class="c5 c6"
      kind="default"
      style="size: default;"
    >
      <span
        class="c7"
      >
        Send
      </span>
    </button>
  </div>
</div>
`;
